packages/core/functions/src/types.ts

```diff
@@ -12,56 +16,48 @@ import { RawObject, S, TypedObject } from '@dxos/echo-schema';
  */
 export type FunctionTriggerType = 'subscription' | 'timer' | 'webhook' | 'websocket';
 
-const SubscriptionTriggerSchema = S.mutable(
-  S.Struct({
-    type: S.Literal('subscription'),
-    // TODO(burdon): Define query DSL (from ECHO).
-    filter: S.Array(
-      S.Struct({
-        type: S.String,
-        props: S.optional(S.Record({ key: S.String, value: S.Any })),
-      }),
-    ),
-    options: S.optional(
-      S.Struct({
-        // Watch changes to object (not just creation).
-        deep: S.optional(S.Boolean),
-        // Debounce changes (delay in ms).
-        delay: S.optional(S.Number),
-      }),
-    ),
-  }),
-);
+const SubscriptionTriggerSchema = S.Struct({
+  type: S.Literal('subscription'),
+  // TODO(burdon): Define query DSL (from ECHO).
```

Comment:
Use query DSL (from ECHO)

Reply:
where is that defined -- need to chat about this. also needed for triggers.